tools/gsc_high_dimensions_visualisation/high_dim_visu.R

```diff
@@ -355,11 +359,13 @@ plot(res.hcpc, choice="map")
 
 # user contrasts on the pca
 if (opt$factor != '') {
+  if(is.factor(contrasting_factor$factor)) {
     plot(pca, label="none", habillage="ind", col.hab=factor_cols)
     legend(x = 'topright', 
        legend = as.character(factorColors$factor),
        col = factorColors$color, pch = 16, bty = 'n', xjust = 1, cex=0.7)
-    }
+  }
```

Comment:
I think an else statement has been forgotten here from the beginning... unless we decide to drop completely the PCA in hcpc, which is/should be a duplication of the PCA option

Reply:
I cannot find how to add a legend for numerical vector for `plot`, there is a package `auto-image` with a function `legend.scale` but there is no conda package.
Unless we use ggplot for the PCA and HCPC plots.

Reply:
If we limit the issue to _PCA_ plots I think using ggplot is perfectly doable !

Reply:
Otherwise, probably some interesting material here https://stackoverflow.com/questions/13355176/gradient-legend-in-base

Reply:
and also https://aurelienmadouasse.wordpress.com/2012/01/13/legend-for-a-continuous-color-scale-in-r/

Reply:
and for ggplot visualisation of pca: https://gist.github.com/benmarwick/2139672.

Reply:
as well as http://www.sthda.com/french/articles/38-methodes-des-composantes-principales-dans-r-guide-pratique/73-acp-analyse-en-composantes-principales-avec-r-l-essentiel/

Reply:
Merci, je regarde tout ça